fcrepo-http-api/src/main/java/org/fcrepo/http/api/ContentExposingResource.java

```diff
@@ -539,10 +564,28 @@ private static void evaluateRequestPreconditions(final Request request,
             return;
         }
 
-        final EntityTag etag = new EntityTag(resource.getEtagValue());
-        final Date date = resource.getLastModifiedDate();
+        final EntityTag etag;
+        final Date date;
         final Date roundedDate = new Date();
 
+        // See the related note about the next block of code in the
+        // ContentExposingResource::addCacheControlHeaders method
+        if (resource instanceof NonRdfSourceDescription) {
+            final NonRdfSourceDescription description = (NonRdfSourceDescription)resource;
+            // Use a weak ETag for the LDP-RS
+            etag = new EntityTag(description.getDescribedResource().getEtagValue(), true);
+            date = description.getDescribedResource().getLastModifiedDate();
+        } else if (resource instanceof NonRdfSource) {
+            final NonRdfSource binary = (NonRdfSource)resource;
+            // Use a strong ETag for the LDP-NR
+            etag = new EntityTag(binary.getDescription().getEtagValue());
```

Comment:
No, you have it backwards. This is the binary: `resource instanceof NonRdfSource`. The confusing logic here is described in the javadoc comment.